### PR TITLE
[Enhancement] Route attribution through native semantic adapters

### PR DESCRIPTION
**Pull request by stukid**
## Summary

- route `attribution_analyze` through the semantic adapter's native `attribute()` contract when available
- retain a normalized generic `term_wise` fallback for adapters without native attribution
- adopt the shared attribution request/result models and structured Dosi validation errors
- update AskMetrics guidance and CLI tool summaries for `term_wise`, `mix_shift`, `factor_shapley`, and `unsupported`
- add real Agent -> Dosi adapter -> native engine -> DuckDB acceptance/nightly coverage

## User-visible behavior

AskMetrics can now return the native Dosi attribution strategy and decomposition instead of applying one generic grouping algorithm to every metric. Native unsupported metrics remain successful tool results with a structured reason; invalid requests retain actionable semantic error codes.

## Validation

- 738 affected unit and integration tests passed
- 12 real Dosi/DuckDB attribution scenarios passed under both acceptance and nightly markers
- Ruff check and format check passed for all changed Python files
- integration test audit passed with zero findings
- `git diff --check`
- manual CLI smoke test: a natural-language January/February net-change question automatically routed to AskMetrics and returned `implementation=dosi`, `strategy=term_wise`

## Dependency

- Companion semantic-adapter contract: Datus-ai/datus-semantic-adapter#100
- Requires the new contract to be published as `datus-semantic-core>=0.2.5`
- The dependency declaration is updated, but `uv.lock` is intentionally unchanged because the package index currently exposes only 0.2.4. Refreshing the lock is required after 0.2.5 is published; `uv lock --check` is expected to remain blocked until then.

## Backport

- [ ] release/0.3.1
- [ ] release/0.3.2
- [ ] release/0.3.3
- [ ] release/0.3.4
- [ ] release/0.3.5
- [ ] release/0.3.6
- [ ] release/0.3.7
- [ ] release/0.3.8
- [ ] release/0.3.9
- [ ] release/0.4.0
- [ ] release/0.4.1
## Summary by CodeRabbit

- **New Features**
  - Added attribution analysis support for term-wise, mix-shift, and factor-based analysis strategies.
  - Added structured attribution results with totals, dimension contributions, drill-down details, reconciliation data, and clearer warnings.
  - Added support for native semantic adapter attribution, parameterized analysis, and improved handling of unsupported metrics and dimensions.
  - Added validation feedback for invalid time windows, missing dimensions, unknown metrics, and invalid parameters.

- **Documentation**
  - Updated semantic tool documentation with attribution models and availability.

## Summary by CodeRabbit

- **New Features**
  - Added attribution analysis for term-wise, mix-shift, and factor-based strategies.
  - Added structured results with totals, dimension contributions, drill-down details, reconciliation data, and clearer warnings.
  - Added native adapter attribution with generic fallback, parameterized analysis, and support for unsupported metrics or dimensions.
  - Added validation feedback for invalid time windows, missing dimensions, unknown metrics, and invalid parameters.

- **Documentation**
  - Updated semantic tool documentation with attribution models and availability.

- **Tests**
  - Added comprehensive integration coverage for attribution workflows and edge cases.

### CODE REVIEW

File: tests/unit_tests/schemas/test_tool_summary.py [hunk 1]
```diff
@@ -476,37 +478,37 @@ def test_attribution_analyze_partial_dimensions(self):
             {
                 "success": 1,
                 "result": {
-                    "candidate_dimensions": ["organization", "channel", "product"],
+                    "implementation": "generic",
+                    "strategy": "term_wise",
                     "per_dimension": {
-                        "organization": {"error": {"code": "DIMENSION_QUERY_FAILED"}},
-                        "channel": {"contributions": []},
+                        "channel": {"values": [], "non_additive": False},
```
[the remark this answers is not in the document]
Comment (stukid) [reply tests/unit_tests/schemas/test_tool_summary.py]: Fixed in 588a63b2c. The tool-summary fixture now includes a `non_additive=True` dimension and asserts the compact summary reports the non-additive count while preserving the existing usable, failed, truncated, and warning counts.

Validation: `uv run pytest -q tests/unit_tests/schemas/test_tool_summary.py`

File: datus/tools/func_tool/attribution_utils.py [hunk 2]
```diff
@@ -118,79 +82,59 @@ def __init__(self, payload: AttributionValidationErrorPayload):
...
             dimensions=[],
-            path=path,
-            time_start=baseline_start,
-            time_end=baseline_end,
-            where=where,
+            path=request.path,
+            time_start=request.baseline.start,
+            time_end=request.baseline.end,
+            where=request.where_sql,
+            params=request.params or None,
```
[the remark this answers is not in the document]
Comment (stukid) [reply datus/tools/func_tool/attribution_utils.py]: Fixed in 588a63b2c. The generic fallback now verifies an explicit `time_dimension` against the adapter's advertised primary time axis before querying. A non-primary or unverifiable axis returns structured `time_dimension_not_supported` with zero metric queries, and only a verified primary axis is recorded in `comparison_metadata`. Added positive and negative unit coverage.

Validation: `uv run pytest -q tests/unit_tests/tools/func_tool/test_attribution_utils.py`

File: pyproject.toml [hunk 3]
```diff
@@ -81,7 +81,7 @@ dependencies = [
     "lxml>=5.0.0",
     "markdown-it-py>=3.0.0",
     "tabulate>=0.9.0",
-    "datus-semantic-core>=0.2.4",
+    "datus-semantic-core>=0.2.5",
```
[the remark this answers is not in the document]
Comment (stukid) [reply pyproject.toml]: Fixed in 588a63b2c. `datus-semantic-core` 0.2.5 is now published, and this commit regenerates `uv.lock` to resolve 0.2.5 while synchronizing `requirements.txt` and `requirements-test.txt`.

Validation: `uv sync --locked --group ci`; `uv lock --check`; `uv run python ci/check_release_readiness.py --repo-root .`
